packages/core/src/errors.ts

```diff
@@ -35,3 +35,36 @@ export class SessionError extends RetryRequestError {
         super(`Detected a session error, rotating session... ${message ? `\n${message}` : ''}`);
     }
 }
+
+export class ContextPipelineInterruptedError extends Error {
+    constructor(message?: string) {
+        super(`Request handling was interrupted during context initialization ${message ? ` - ${message}` : ''}`);
+    }
+}
+
+export class ContextPipelineInitializationError extends Error {
+    constructor(
+        public error: unknown,
```

Comment:
Perhaps you'd want to use `Error { cause }` field [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause)? It seems more idiomatic that way.